src/state/zustand/swapperStore/amountSelectors.ts

```diff
@@ -588,3 +590,11 @@ export const selectDonationAmountFiat = createSelector(
     return bnOrZero(sellAmountFiat).times(affiliatePercentage).toFixed()
   },
 )
+
+export const selectFailureAmounts = createDeepEqualOutputSelector(
+  (state: SwapperState) => state.activeSwapperWithMetadata?.quote?.failureAmounts,
+  (state: SwapperState) => state.trade?.failureAmounts,
+  (quoteFailureAmounts, tradeFailureAmounts): FailureAmount[] | undefined =>
+    // Use the trade amount if we have it, otherwise use the quote amount
```

Comment:
Is this effectively selecting actual failed amounts, or intermediary steps?

Reply:
its an array of all of the possible tokens you could end up with once you start executing a trade. Generally with lifi there are 1-2 intermediary tokens the user may end up with in the event of a failure on lifi side

Reply:
ill improve the comment in `src/lib/swapper/api.ts` :)